pkg/agent/nodeportlocal/k8s/npl_controller.go

```diff
@@ -117,6 +119,14 @@ func podKeyFunc(pod *corev1.Pod) string {
 	return pod.Namespace + "/" + pod.Name
 }
 
+func (c *NPLController) Initialize() error {
+	klog.InfoS("Will fetch Pods and generate NodePortLocal rules for these Pods")
+	if err := c.GetPodsAndGenRules(); err != nil {
```

Comment:
After it's extracted to a method and called separately, it doesn't wait for cache to sync, I think it may miss Pod data?